addons/mail/models/mail_thread.py

```diff
@@ -3373,7 +3373,15 @@ def _message_update_content_after_hook(self, message):
 
     def _get_mail_thread_data_attachments(self):
         self.ensure_one()
-        return self.env['ir.attachment'].search([('res_id', '=', self.id), ('res_model', '=', self._name)], order='id desc')
+        res = self.env['ir.attachment'].search([('res_id', '=', self.id), ('res_model', '=', self._name)], order='id desc')
+        if 'original_id' in self.env['ir.attachment']._fields:
+            # If the image is SVG: We take the png version if exist otherwise we take the svg
+            # If the image is not SVG: We take the original one if exist otherwise we take it
+            svg_ids = set(res.filtered(lambda attachment: attachment.mimetype == 'image/svg+xml').ids)
+            non_svg_ids = set(res.ids) - svg_ids
+            original_ids = set(res.mapped('original_id').ids)
+            res = res.filtered(lambda attachment: (attachment.id in svg_ids and attachment.id not in original_ids) or (attachment.id in non_svg_ids and attachment.original_id.id not in non_svg_ids))
```

Comment:
recordsets are supporting set operations, I don't think you need to manipulate set of ids, the code could be much simpler with just working with recordsets.